play-services-core/src/main/java/org/microg/gms/gcm/McsService.java

```diff
@@ -485,6 +487,26 @@ private void handleAppMessage(DataMessageStanza msg) {
             for (ResolveInfo resolveInfo : infos) {
                 logd("Target: " + resolveInfo);
                 Intent targetIntent = new Intent(intent);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Another minor: it would make sense to use the `app.wakeForDelivery` setting as a restriction whether to add the app to the temp whitelist or not, no?

Reply:
That sounds reasonable to me.  The current description of this option sounds like what the whitelist is intended to do.

> Start the app in background to receive the incoming push messages. Disabling this might delay message delivery up to the moment the app is started or cause complete message loss.

Currently it seems like this option was only used to send the broadcast to [an app that was stopped](https://developer.android.com/reference/android/content/Intent.html#FLAG_INCLUDE_STOPPED_PACKAGES).